clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -281,13 +281,15 @@ check_sane_control_flow()
             gen_marker(TID, TRACE_MARKER_TYPE_FILETYPE, OFFLINE_FILE_TYPE_ENCODINGS),
 #    if defined(X86_64) || defined(X86_32)
             // 0x74 is "je" with the 2nd byte the offset.
-            gen_branch_encoded(TID, 0x71019dbc, { 0x74, 0x32 }),
-            gen_instr_encoded(0x71019ded, { 0x01 }, TID),
+            gen_branch_encoded(TID, /*pc=*/0x71019dbc, /*encoding=*/ { 0x74, 0x32 }),
```

Comment:
The statement

gen_instr_encoded(0x71019ded, { 0x01 }, TID)

causes an expectation of one read and one write memory access. Since 0x01 is invalid, I changed it to nop to proceed.

I can investigate why it's the case. Since it happens only in test, I'd consider it low priority. Please let me know if you think otherwise.

Reply:
OK, I see.  I think I didn't understand the PR description mentioning this: maybe reword that?

Reply:
Eventually we'd like to replace these with an instrlist too; could be done now.

Reply:
I've updated the code to use instrlist here.